### ISSUE TITLE
Bad behavior with description lists

### ISSUE THREAD

**Issue by aecay** [calvinwyoung/org-autolist#2]
This library has odd behavior when hitting backspace on a description list with an empty description.  To reproduce, enter the following text (where ! represents the point) in a buffer with org mode and org-autolist activated, and hit backspace.

```
- foo :: bar
- quux :: !
```

The result is that the second list item is entirely deleted.  I would have expected to be allowed to backspace over the string "quux :: " before hitting the beginning of the list item, at which point org-autolist's automgic backspace would trigger.  This also happens if the point is at the beginning of a non-empty list item: 

```
- foo :: bar
- quux :: !abc
```

yields when backspacing:

```
- foo :: bar!abc
```

**Comment by calvinwyoung** [comment]
@aecay Thanks for the bug report, and sorry for the delayed response. I'm looking into this right now, and I'll follow up when I have a good solution.

**Comment by calvinwyoung** [comment]
@aecay I finally had a chance to look at this over the weekend. These issues should be fixed in #3. Let me know if you run into any other issues w/ org-autolist!

**Comment by aecay** [comment]
Thanks!
